e2e/app/page/notebook-page.ts

```diff
@@ -115,30 +115,59 @@ export default class NotebookPage extends NotebookFrame {
 
   async openUploadFilePage(): Promise<Page> {
     // Select File menu => Open.
-    await this.selectFileOpenMenu();
-
-    // New tab opens. "browser" is a Jest-Puppeteer global variable.
-    const newTarget = await browser.waitForTarget((target) => target.opener() === this.page.target());
-    const newPage = await newTarget.page();
+    const newPage = await this.selectFileOpenMenu();
 
     // Upload button that triggers file selection dialog.
     const uploadButtonSelector = 'input#upload_span_input';
     await newPage.waitForSelector(uploadButtonSelector, { visible: true });
     return newPage;
   }
 
-  async acceptDataUsePolicyDialog(page: Page): Promise<void> {
+  async downloadFileFromTree(treePage: Page, filename: string): Promise<Page> {
+    // Check the box for the given filename and download.
+    const findAndClickCheckbox = () =>
+      treePage
+        .waitForXPath(
+          `//*[@id="notebook_list"]//a[@class="item_link"][span[text()="${filename}"]]/preceding-sibling::input`,
+          { visible: true }
+        )
+        .then((e) => e.click());
+    await findAndClickCheckbox();
+
+    const handleDialog = (d) => this.acceptDataUseDownloadDialog(treePage, d);
```

Comment:
You need the `async` keyword if you want to use `await` in your function, that's it. In this case, there is no need to use the await keyword, therefore we don't need it to be an async function. What you've written is functionally equivalent to what I wrote, but with extra steps.

It will be difficult to explain this in a comment - lets stay on and talk through it after eng sync, if you want.